src/test/java/marquez/service/NamespaceServiceTest.java

```diff
@@ -1,132 +1,163 @@
+import marquez.service.mappers.NamespaceMapper;
 import marquez.service.models.Namespace;
 import org.jdbi.v3.core.statement.UnableToExecuteStatementException;
-import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
 
+@Category(UnitTests.class)
 public class NamespaceServiceTest {
-  private final NamespaceName NAMESPACE_NAME = NamespaceName.fromString("test");
-
-  private static final NamespaceDao namespaceDao = mock(NamespaceDao.class);
-  NamespaceService namespaceService = new NamespaceService(namespaceDao);
+  private final NamespaceName NAMESPACE_NAME = newNamespaceName();
+  private final Description DESCRIPTION = newDescription();
+  private final OwnerName CURRENT_OWNER_NAME = newOwnerName();
+  private final Namespace NEW_NAMESPACE =
+      new Namespace(
+          null, NAMESPACE_NAME.getValue(), CURRENT_OWNER_NAME.getValue(), DESCRIPTION.getValue());
+
+  private NamespaceDao namespaceDao;
+  private NamespaceService namespaceService;
+
+  @Before
+  public void setUp() throws MarquezServiceException {
+    namespaceDao = mock(NamespaceDao.class);
+    namespaceService = new NamespaceService(namespaceDao);
+  }
 
-  @After
-  public void teardown() {
-    reset(namespaceDao);
+  @Test
+  public void testNewDatasetService_throwsException_onNullNamespaceDao() {
+    final NamespaceDao nullNamespaceDao = null;
+    assertThatNullPointerException().isThrownBy(() -> new NamespaceService(nullNamespaceDao));
   }
 
   @Test
   public void testCreate() throws MarquezServiceException {
-    Namespace ns = Generator.genNamespace();
-    final NamespaceName namespaceName = NamespaceName.fromString(ns.getName());
-    when(namespaceDao.findBy(namespaceName))
-        .thenReturn(
-            Optional.of(
-                new Namespace(
-                    UUID.randomUUID(), ns.getName(), ns.getOwnerName(), ns.getDescription())));
-    Namespace nsReturned = namespaceService.create(ns);
-    verify(namespaceDao).insert(any(Namespace.class));
-    assertEquals(ns.getName(), nsReturned.getName());
-    assertEquals(ns.getOwnerName(), nsReturned.getOwnerName());
-    assertEquals(ns.getDescription(), nsReturned.getDescription());
-  }
+    reset(namespaceDao);
+    namespaceService.create(NEW_NAMESPACE);
 
-  @Test(expected = MarquezServiceException.class)
-  public void testCreate_findException() throws MarquezServiceException {
-    Namespace ns = Generator.genNamespace();
-    doThrow(UnableToExecuteStatementException.class).when(namespaceDao).findBy(NAMESPACE_NAME);
-    namespaceService.create(ns);
+    verify(namespaceDao, times(1)).insert(any(NamespaceRow.class));
   }
 
-  @Test(expected = MarquezServiceException.class)
-  public void testCreate_insertException() throws MarquezServiceException {
-    Namespace ns = Generator.genNamespace();
+  @Test
+  public void testCreate_throwsException_onDbError() throws MarquezServiceException {
+    reset(namespaceDao);
```

Comment:
why the reset here?

Reply:
A reset of the `NamespaceDao` mock is needed because `NamespaceService` now has an [`init()`](https://github.com/MarquezProject/marquez/blob/feature/dataset-resource-create-get-ops/src/main/java/marquez/service/NamespaceService.java#L41) methods which calls out to the DB in order to create the _default_ namespace.

Meaning, for the test to pass we would need the following mock verification (note the `2` count):

```java
verify(namespaceDao, times(2)).insert(any(NamespaceRow.class));
```

instead of:

```java
verify(namespaceDao, times(1)).insert(any(NamespaceRow.class));
```

_To sum up_: the reset allows us to verify the insert was the result of the service (**after** instantiation) performing a DAO call instead of **on** service instantiation.

Reply:
Probably worth commenting on why!